### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mihutmalin**
Added:
- Chapter 1
- Chapter 2
- Chapter 3
- Chapter 4
Mentions: 
- The way I implemented the strategies is not exactly classy but it works
- The methods are tested in Postman

### CODE REVIEW

File: shop/src/main/java/ro/msg/learning/shop/Configurations/StrategyConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import ro.msg.learning.shop.Strategies.SingleLocationStrategyConfiguration;
+import ro.msg.learning.shop.Strategies.StrategyConfigurationInterface;
+
+@Configuration
+@AllArgsConstructor
+public class StrategyConfiguration {
+
+    @Bean
+    public StrategyConfigurationInterface chooseStrategy(@Value("${strategy}")String s) throws Exception {
+        if(s.equals("SingleLocationStrategy")){
```
Comment (cristinacalbaza) [inline shop/src/main/java/ro/msg/learning/shop/Configurations/StrategyConfiguration.java]: You can create an Enum for Strategy Type and replace the hardcoded text with Enum value

File: shop/src/main/java/ro/msg/learning/shop/ShopApplication.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	public static void main(String[] args) {
+		SpringApplication.run(ShopApplication.class, args);
+		/*
+		ApplicationContext ctx = new AnnotationConfigApplicationContext(StrategyConfiguration.class);
+
+		StrategyConfiguration strategy = ctx.getBean(StrategyConfiguration.class);
+		strategy.doSingleLocation();
+		strategy.doMostAbundant();
+		 */
```
Comment (cristinacalbaza) [inline shop/src/main/java/ro/msg/learning/shop/ShopApplication.java]: Remove this if it is not needed.

**Review by cristinacalbaza — CHANGES_REQUESTED** [review]

File: shop/src/main/java/ro/msg/learning/shop/Configurations/StrategyConfiguration.java (hunk 1 above)
Comment (mihutmalin) [reply shop/src/main/java/ro/msg/learning/shop/Configurations/StrategyConfiguration.java]: Fixed.

File: shop/src/main/java/ro/msg/learning/shop/ShopApplication.java (hunk 2 above)
Comment (mihutmalin) [reply shop/src/main/java/ro/msg/learning/shop/ShopApplication.java]: Fixed.

**Review by cristinacalbaza — APPROVED** [review]

**Review by cristinacalbaza — APPROVED** [review]

**Pull request closed without merging**